File: frontend/web/src/types/budget.ts
// src/types/budget.ts
export type BudgetPeriodType = "Monthly" | "Weekly";

export interface Budget {
  id: number;
  userId: number;
  category: string;
  periodType: number; // 0=Monthly,1=Weekly (EF enum)
  limitAmount: number;
  isActive: boolean;
  createdAt: string;
  lastAlertSentAt?: string | null;
}

File: backend/api/Models/Budget.cs
// Models/Budget.cs
namespace SmartCampusWallet.Api.Models;

public enum BudgetPeriodType
{
    Monthly = 0,
    Weekly = 1
}

public class Budget
{
    public int Id { get; set; }

    public int UserId { get; set; }

    // e.g. "Dining", "Transport", "Books", "Events", "Groceries"
    public string Category { get; set; } = null!;

    public BudgetPeriodType PeriodType { get; set; } = BudgetPeriodType.Monthly;

    public decimal LimitAmount { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // for 80% notifications – we only want to send once per period
    public DateTime? LastAlertSentAt { get; set; }
    
    // Track if 100% alert has been sent this period
    public bool OverBudgetAlertSent { get; set; } = false;
}
